### ISSUE TITLE
Fails to deserialize array

### ISSUE THREAD

**Issue by xplicit** [salarcode/Bois#1]
This code is failed to deserialize

```
[Serializable]
public class ByteArray64K
{
    public byte[] Arr { get; set; }
}


class MainClass
{
    public static void Main (string[] args)
    {
        var ser = new BoisSerializer ();

        var arr = new ByteArray64K(){Arr = new byte[65536]};
        byte[] data;

        using (MemoryStream ms = new MemoryStream ()) {
            ser.Serialize (arr, ms);
            data = ms.ToArray ();
        }

        ByteArray64K des1;

        using (MemoryStream ms = new MemoryStream (data)) {
            des1=ser.Deserialize<ByteArray64K>(ms);
        }

        Console.WriteLine ("length1={0}, length2={1}", arr.Arr.Length, des1.Arr.Length);
    }
}
```

Result is: length1=65536, length2=0.

bois version 1.8
mono --version

```
Mono JIT compiler version 3.10.0 (tarball Wed Nov  5 13:32:50 UTC 2014)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
TLS:           __thread
SIGSEGV:       altstack
Notifications: epoll
Architecture:  x86
Disabled:      none
Misc:          softdebug 
LLVM:          supported, not enabled.
GC:            sgen
```

**Comment by salarcode** [comment]
Thanks for the quick report.
The bug was related to the array length you have used, ushort+1 and I believe it was a serious bug.
It is fixed in 1.8.1.

**Comment by xplicit** [comment]
Thank you,

I confirm, it's working now.
